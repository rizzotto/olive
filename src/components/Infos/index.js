import React, { useState } from 'react';
import useStyles from './styles';
import PropTypes from 'prop-types';
import Input from '../Input';

function Infos({
  onChangeTime,
  onChangePortions,
  defaultPortions,
  defaultTime,
}) {
  const classes = useStyles();
  const [symbolsArr] = useState(['e', 'E', '+', '-', '.']);

  function handleTextChange(e, callback) {
    e.target.value = e.target.value.replace(/-/g, '');
    callback(e.target.value);
  }

  return (
    <div className={classes.infos}>
      <Input
        className={classes.inputs}
        type="number"
        hiddenLabel
        placeholder="60"
        inputProps={{ min: 0, style: { textAlign: 'center' } }}
        onChange={e => handleTextChange(e, onChangeTime)}
        onKeyDown={e => symbolsArr.includes(e.key) && e.preventDefault()}
        defaultValue={defaultTime === isNaN ? '' : defaultTime}
        maxSize={3}
        onInput={e => {
          e.target.value = Math.max(0, parseInt(e.target.value))
            .toString()
            .slice(0, 3);
        }}
      />
      <span> Minutos </span>

      <Input
        className={classes.inputs}
        onInput={e => {
          e.target.value = Math.max(0, parseInt(e.target.value))
            .toString()
            .slice(0, 3);
        }}
        type="number"
        hiddenLabel
        placeholder="7"
        inputProps={{ min: 0, style: { textAlign: 'center' } }}
        onKeyDown={e => symbolsArr.includes(e.key) && e.preventDefault()}
        onChange={e => handleTextChange(e, onChangePortions)}
        defaultValue={defaultPortions === isNaN ? '' : defaultPortions}
        maxSize={3}
      />
      <span> Porções </span>
    </div>
  );
}

Infos.propTypes = {
  onChangeTime: PropTypes.func,
  onChangePortions: PropTypes.func,
};

Infos.defaultProps = {
  onChangeTime: () => {},
  onChangePortions: () => {},
};

export default Infos;
