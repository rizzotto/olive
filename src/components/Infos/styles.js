import { makeStyles } from '@material-ui/styles';

const useStyles = makeStyles(() => ({
  infos: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'flex-start',
    '& span': {
      margin: '0px 10px',
    },
  },
  inputs: {
    width: '60px',
  },
}));

export default useStyles;
