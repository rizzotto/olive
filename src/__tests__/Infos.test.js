import React from 'react';
import renderer from 'react-test-renderer';
import Infos from '../components/Infos';

test('<Infos /> should render with no props', () => {
  const tree = renderer
    .create(<Infos />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('<Infos /> should render with props', () => {
    const tree = renderer
      .create(<Infos onChangePortions={() => {}} onChangeTime={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  