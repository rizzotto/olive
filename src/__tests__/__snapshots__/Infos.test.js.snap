// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Infos /> should render with no props 1`] = `
<div
  className="makeStyles-infos-1"
>
  <div
    className="MuiFormControl-root MuiTextField-root makeStyles-inputs-2 MuiFormControl-fullWidth"
  >
    <div
      className="MuiInputBase-root MuiFilledInput-root makeStyles-input-3 MuiInputBase-fullWidth MuiInputBase-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputHiddenLabel MuiFilledInput-inputHiddenLabel"
        disabled={false}
        min={0}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="60"
        required={false}
        rows={1}
        style={
          Object {
            "textAlign": "center",
          }
        }
        type="number"
      />
    </div>
  </div>
  <span>
     Minutos 
  </span>
  <div
    className="MuiFormControl-root MuiTextField-root makeStyles-inputs-2 MuiFormControl-fullWidth"
  >
    <div
      className="MuiInputBase-root MuiFilledInput-root makeStyles-input-3 MuiInputBase-fullWidth MuiInputBase-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputHiddenLabel MuiFilledInput-inputHiddenLabel"
        disabled={false}
        min={0}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="7"
        required={false}
        rows={1}
        style={
          Object {
            "textAlign": "center",
          }
        }
        type="number"
      />
    </div>
  </div>
  <span>
     Porções 
  </span>
</div>
`;

exports[`<Infos /> should render with props 1`] = `
<div
  className="makeStyles-infos-1"
>
  <div
    className="MuiFormControl-root MuiTextField-root makeStyles-inputs-2 MuiFormControl-fullWidth"
  >
    <div
      className="MuiInputBase-root MuiFilledInput-root makeStyles-input-3 MuiInputBase-fullWidth MuiInputBase-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputHiddenLabel MuiFilledInput-inputHiddenLabel"
        disabled={false}
        min={0}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="60"
        required={false}
        rows={1}
        style={
          Object {
            "textAlign": "center",
          }
        }
        type="number"
      />
    </div>
  </div>
  <span>
     Minutos 
  </span>
  <div
    className="MuiFormControl-root MuiTextField-root makeStyles-inputs-2 MuiFormControl-fullWidth"
  >
    <div
      className="MuiInputBase-root MuiFilledInput-root makeStyles-input-3 MuiInputBase-fullWidth MuiInputBase-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputHiddenLabel MuiFilledInput-inputHiddenLabel"
        disabled={false}
        min={0}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="7"
        required={false}
        rows={1}
        style={
          Object {
            "textAlign": "center",
          }
        }
        type="number"
      />
    </div>
  </div>
  <span>
     Porções 
  </span>
</div>
`;
